src/lib/linux-api/src/signal.rs

```diff
@@ -0,0 +1,548 @@
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+use vasi::VirtualAddressSpaceIndependent;
+
+use crate::bindings;
+
+pub const LINUX_STANDARD_SIGNAL_MAX_NO: i32 = 31;
+
+/// Lowest realtime signal number.
+pub const LINUX_SIGRT_MIN: i32 = bindings::SIGRTMIN as i32;
+/// Highest realtime signal number.
+//
+// According to signal(7). bindgen fails to bind this one.
+pub const LINUX_SIGRT_MAX: i32 = 64;
+
+/// Definition is sometimes missing in the userspace headers.
+//
+// bindgen fails to bind this one.
+// Copied from linux's include/uapi/linux/signal.h.
+pub const LINUX_SS_AUTODISARM: i32 = 1 << 31;
+
+// Bindgen doesn't succesfully bind these constants; maybe because
+// the macros defining them cast them to pointers.
+//
+// Copied from linux's include/uapi/asm-generic/signal-defs.h.
+pub const LINUX_SIG_DFL: usize = 0;
+pub const LINUX_SIG_IGN: usize = 1;
+pub const LINUX_SIG_ERR: usize = (-1_isize) as usize;
+
+// signal names
+#[derive(Debug, Copy, Clone, IntoPrimitive, TryFromPrimitive)]
+#[repr(i32)]
+pub enum LinuxSignal {
```

Comment:
This is now a struct, so that it can cleanly represent realtime signals. While making other changes I noticed we were doing an unsound transmute to work around nix's Signal not explicitly representing them.